daemon/src/engine/server/CryptoChallege.h

```diff
@@ -0,0 +1,633 @@
+/*
```

Comment:
Also we don't need to choose an algorithm at runtime as we own both the server, the client and the tools. If some day we need to choose an algorithm at runtime, then we'll introduce the required complexity.

The trivial methods I was speaking about are Data's methods.